orchestrator/careplancontributor/handle_taskfiller_test.go

```diff
@@ -3,171 +3,257 @@ package careplancontributor
 		profile          profile.Provider
-		task             *fhir.Task
-		expectedError    bool
+		notificationTask fhir.Task
+		primaryTask      *fhir.Task
+		serviceRequest   fhir.ServiceRequest
+		expectedError    error
 		numBundlesPosted int
 		mock             func(*mock.MockClient)
 	}{
 		{
-			name: "Valid SCP Task",
-			ctx:  auth.WithPrincipal(context.Background(), *auth.TestPrincipal1),
-			profile: profile.TestProfile{
-				Principal: auth.TestPrincipal1,
-			},
-			task:             &primaryTask,
-			expectedError:    false,
+			name:             "primary task, owner = local organization, triggers subtask creation",
+			notificationTask: deep.Copy(primaryTask),
 			numBundlesPosted: 1, // One subtask should be created
 		},
 		{
-			name: "Valid SCP Task - Not owner",
-			profile: profile.TestProfile{
-				Principal: auth.TestPrincipal2,
-			},
+			name:             "primary task, owner != local organization, nothing should happen",
+			profile:          profile.TestProfile{Principal: auth.TestPrincipal2},
 			ctx:              auth.WithPrincipal(context.Background(), *auth.TestPrincipal2),
-			task:             &primaryTask,
-			expectedError:    false,
-			numBundlesPosted: 0,
+			notificationTask: deep.Copy(primaryTask),
 		},
 		{
-			name: "Invalid Task (Missing Profile)",
-			profile: profile.TestProfile{
-				Principal: auth.TestPrincipal1,
+			name: "primary task, contains reasonReference instead of reasonCode, triggers subtask creation",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.ReasonReference = &fhir.Reference{
+					Reference: to.Ptr("Condition/1"),
+				}
+				t.ReasonCode = nil
+			}),
+			mock: func(client *mock.MockClient) {
+				client.EXPECT().Read("Condition/1", gomock.Any(), gomock.Any()).
+					DoAndReturn(func(_ string, result *fhir.Condition, _ ...fhirclient.Option) error {
+						*result = fhir.Condition{
+							Id: to.Ptr("1"),
+							Code: &fhir.CodeableConcept{
+								Coding: []fhir.Coding{
+									{
+										System: to.Ptr("http://snomed.info/sct"),
+										Code:   to.Ptr("13645005"),
+									},
+								},
+							},
+						}
+						return nil
+					})
+			},
+			numBundlesPosted: 1, // One subtask should be created
+		},
+		{
+			name: "error: primary task, multiple reasonCodes match",
+			notificationTask: deep.AlterCopy(primaryTask, func(t *fhir.Task) {
+				t.ReasonCode = &fhir.CodeableConcept{
+					Coding: []fhir.Coding{
+						{
+							System: to.Ptr("http://snomed.info/sct"),
+							Code:   to.Ptr("13645005"), // COPD
+						},
+						{
+							System: to.Ptr("http://snomed.info/sct"),
+							Code:   to.Ptr("84114007"), // Heart failure
+						},
+					},
+				}
+			}),
+			expectedError: errors.New("failed to process new primary Task: Task.reasonCode or Task.reasonReference matches multiple workflows (http://snomed.info/sct|13645005, http://snomed.info/sct|84114007) for service http://snomed.info/sct|719858009"),
+		},
```

Comment:
Might be good to also add a case that proves duplicate codes (two COPD) are fine, as long as it results in one reasonCodes